### PR TITLE
Allocations, Deposits, updated Exchange endpoints

### PR DESCRIPTION
**Pull request by jprix**

### CODE REVIEW

File: src/UI/Orders/AllocationsList.svelte [hunk 1]
```diff
@@ -0,0 +1,280 @@
+<script>
+    /**
+     * Copyright 2022 Coinbase Global, Inc.
```
Comment (rcbgr) [inline src/UI/Orders/AllocationsList.svelte:3]: Add 2023 for new files

File: src/UI/Orders/AllocationsList.svelte [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    import _ from 'lodash-es';
+    import { getDate } from '../../utils/constants'
+
+    let allocations;
+    let allocationsForm = false;
+    let allocationsFilterView = false;
+    let FilteredAllocations;
+    let queryParams;
+    let loaded;
+    const startDate= getDate(new Date(), -6);
```
Comment (rcbgr) [inline src/UI/Orders/AllocationsList.svelte:37]: Space issue

File: src/UI/Orders/AllocationsList.svelte [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  
+    const { form, handleChange, handleSubmit } = createForm({
+      initialValues: {
+        product_ids: 'BTC-USD',
+        start_date: startDate,
+      },
+      onSubmit: (values) => {
+        const { product_ids, order_type, cursor, limit, order_side, start_date, end_date } = values;
+        
+
```
Comment (rcbgr) [inline src/UI/Orders/AllocationsList.svelte:82]: Remove extra lines

File: src/UI/UsersList.svelte [hunk 4]
```diff
@@ -23,7 +23,14 @@
     userList = await getPortfolioUsers();
   });
 </script>
-
+<div>
+  <a class="buy-with-crypto"
+    href="https://commerce.coinbase.com/checkout/a5b2403c-94e4-446a-9f1b-822e2b4a8e42">
```
Comment (rcbgr) [inline src/UI/UsersList.svelte:28]: What is this UUID?

File: src/UI/UsersList.svelte [hunk 5]
```diff
@@ -23,7 +23,14 @@
     userList = await getPortfolioUsers();
   });
 </script>
-
+<div>
+  <a class="buy-with-crypto"
```
Comment (rcbgr) [inline src/UI/UsersList.svelte:27]: Why is commerce referenced?

File: src/apis/Allocations.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2022 Coinbase Global, Inc.
```
Comment (rcbgr) [inline src/apis/Allocations.js:2]: 2023

File: src/apis/Allocations.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import _ from 'lodash-es';
+
+ let allocations;
+ let result;
+ 
+ export const getAllocations = async (queryParams) => {
+   const { port, portfolioId, httpHost } = await fetchStore();
+   const path = `/v1/portfolios/${portfolioId}/allocations`;
+   const allocationsUrl = `${httpHost}:${port}/api${path}?${
+    queryParams ? queryParams : 'start_date=2022-06-05T00:00:01Z'
```
Comment (rcbgr) [inline src/apis/Allocations.js:27]: Does this date need to be dynamic?

File: src/apis/Exchange/Currencies.js [hunk 8]
```diff
@@ -6,6 +6,7 @@ export const getCurrencies = async () => {
 
   const url = `${httpHost}:${port}/api/exchange/currencies`;
   const path = '/currencies';
```
Comment (rcbgr) [inline src/apis/Exchange/Currencies.js:8]: Is path reference somewhere else? If not, just add the string to makeCall

File: src/apis/Exchange/Deposits.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import { fetchStore } from '../../stores/userSession-store';
+import { makeCall } from './ExchangeClient';
+//import { getAccounts } from '../../apis/Exchange/coinbase-accounts';
```
Comment (rcbgr) [inline src/apis/Exchange/Deposits.js:3]: Remove unused code

File: src/apis/Exchange/Withdrawals.js [hunk 10]
```diff
@@ -17,7 +17,7 @@ export const WithdrawToCoinbaseAccount = async (
   };
 
   const payload = JSON.stringify(body);
-  const path = `/withdrawals/coinbase-account`;
```
Comment (rcbgr) [inline src/apis/Exchange/Withdrawals.js:20]: Should this be a single quote instead of a backtick?

File: src/apis/Exchange/Withdrawals.js [hunk 11]
```diff
@@ -17,7 +17,7 @@ export const WithdrawToCoinbaseAccount = async (
...
+  const path = `/withdrawals/coinbase-account/`;
```
Comment (rcbgr) [inline src/apis/Exchange/Withdrawals.js:20]: Single quote?

File: src/utils/constants.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+const getDaysInMonth = (year, month) => new Date(year, month, 0).getDate()
+
+
+export const getDate = (input, months) => {
+  const date = new Date(input)
+  date.setDate(1)
```
Comment (rcbgr) [inline src/utils/constants.js:6]: Why is this always 1?

File: src/utils/constants.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+  date.setMonth(date.getMonth() + months)
+  date.setDate(Math.min(input.getDate(), getDaysInMonth(date.getFullYear(), date.getMonth()+1)))
```
Comment (rcbgr) [inline src/utils/constants.js:8]: What happens if the current month is December?

**Review by rcbgr — CHANGES_REQUESTED** [review]
Please squash commits too

File: src/UI/UsersList.svelte (hunk 5 above)
Comment (jprix) [reply src/UI/UsersList.svelte]: I was experimenting and forgot to remove.

**Review by rcbgr — COMMENTED** [review]
Files created in 2022 should be: 2022-present
Files created in 2023 should be: 2023-present

File: src/UI/UsersList.svelte (hunk 4 above)
Comment (jprix) [reply src/UI/UsersList.svelte]: removed commerce reference

File: src/apis/Allocations.js (hunk 7 above)
Comment (jprix) [reply src/apis/Allocations.js]: its a ternary, the user can pass in a dynamic date but they enter nothing in the filter form, i use that date

File: src/apis/Exchange/Currencies.js (hunk 8 above)
Comment (jprix) [reply src/apis/Exchange/Currencies.js:8]: it is, used  as a method param on line 11 to make the function call

**Review by rcbgr — APPROVED** [review]
LGTM

File: src/utils/constants.js (hunk 12 above)
Comment (jprix) [reply src/utils/constants.js]: removed line 6.  not needed.

**Review by nmorgancb — APPROVED** [review]

**Review by cfluke-cb — APPROVED** [review]

**Pull request merged**